pkg/pgmodel/ingestor/trace/writer.go

```diff
@@ -126,14 +126,22 @@ func getServiceName(rSpan pdata.ResourceSpans) string {
 	return serviceName
 }
 
+var (
+	traceLabel      = prometheus.Labels{"type": "trace"}
+	traceSpanLabel  = prometheus.Labels{"type": "trace", "kind": "span"}
+	traceEventLabel = prometheus.Labels{"type": "trace", "kind": "event"}
+	traceLinkLabel  = prometheus.Labels{"type": "trace", "kind": "link"}
+)
+
 func (t *traceWriterImpl) InsertTraces(ctx context.Context, traces pdata.Traces) error {
 	startIngest := time.Now() // Time taken for complete ingestion => Processing + DB insert.
 	code := "500"
-	metrics.IngestorActiveWriteRequests.With(prometheus.Labels{"type": "trace", "kind": "span"}).Inc()
+	metrics.IngestorActiveWriteRequests.With(traceSpanLabel).Inc()
+	metrics.InsertablesReceivedTotal.With(traceSpanLabel).Add(float64(traces.SpanCount()))
```

Comment:
we should not have this, as its already caught by 354 `metrics.IngestorItems.With(traceSpanLabel).Add(float64(traces.SpanCount()))`

Reply:
These two things are not semantically the same. What I wanted is to get number of spans received - doesn't mean they are ingested. Line 354 will only be executed in case there was no failure meaning that we will miss the numbers in case of error.
I will move `IngestorItems` in place where `InsertablesReceivedTotal` was used.
I also noticed that we are not consistent with using `IngestorItems`. For metrics ingestion we increase it even if an error occurs, however for tracing we were not doing that.